akka-distributed-data/src/main/scala/akka/cluster/ddata/ORSet.scala

```diff
@@ -307,6 +333,46 @@ final class ORSet[A] private[akka] (
     }
   }
 
+  private def mergeAddDelta(that: ORSet[A]): ORSet[A] = {
+    val commonKeys =
+      if (this.elementsMap.size < that.elementsMap.size)
+        this.elementsMap.keysIterator.filter(that.elementsMap.contains)
+      else
+        that.elementsMap.keysIterator.filter(this.elementsMap.contains)
+    val entries00 = ORSet.mergeCommonKeys(commonKeys, this, that)
+    // FIXME
+    //    val thisUniqueKeys = this.elementsMap.keysIterator.filterNot(that.elementsMap.contains)
+    //    val entries0 = ORSet.mergeDisjointKeys(thisUniqueKeys, this.elementsMap, that.vvector, entries00)
+    val entries0 = entries00 ++ this.elementsMap.filter { case (elem, _) ⇒ !that.elementsMap.contains(elem) }
+    val thatUniqueKeys = that.elementsMap.keysIterator.filterNot(this.elementsMap.contains)
+    val entries = ORSet.mergeDisjointKeys(thatUniqueKeys, that.elementsMap, this.vvector, entries0)
+    val mergedVvector = this.vvector.merge(that.vvector)
+
+    clearAncestor()
+    new ORSet(entries, mergedVvector)
+  }
+
+  private def mergeRemoveDelta(that: ORSet[A]): ORSet[A] = {
```

Comment:
Please also see comment above on having a possibility of having `deltas` as a list.